diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index 4b3dfa5..c458da9 100755
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -859,7 +859,7 @@ function! fzf#vim#ag_raw(command_suffix, ...)
   if !executable('ag')
     return s:warn('ag is not found')
   endif
-  return call('fzf#vim#grep', extend(['ag --nogroup --column --color '.a:command_suffix, 1], a:000))
+  return call('fzf#vim#grep', extend(['ag --nogroup --color '.a:command_suffix, 1], a:000))
 endfunction
 
 " command (string), [spec (dict)], [fullscreen (bool)]
