diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index 4f7003f..061993a 100644
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -615,7 +615,6 @@ function! fzf#vim#gitfiles(args, ...)
   if a:args != '?'
     return s:fzf('gfiles', {
     \ 'source':  'git ls-files '.a:args.(s:is_win ? '' : ' | uniq'),
-    \ 'dir':     root,
     \ 'options': '-m --prompt "GitFiles> "'
     \}, a:000)
   endif
